packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -202,7 +208,10 @@ export default class Proxy {
 
       await xapi.startVm(vm.$id)
     } catch (error) {
-      await xapi._deleteVm(vm)
+      await Promise.all([
+        app.unbindLicense({ licenseId, boundObjectId: vm.uuid }),
+        xapi._deleteVm(vm),
+      ])
```

Comment:
What if one of this operation fails?

How do we get the original error?

Reply:
We'll not get the original error but the revert operations also needs to be reported.

Reply:
This is an issue because we'll never know what was the initial failures.

Reply:
So we need to construct an error which wrap multiple errors. There have a clean way to do it?